packages/dds/tree/src/core/rebase/types.ts

```diff
@@ -201,10 +201,16 @@ export type ChangeMetadata = CommitMetadata &
 				 * This is a `SerializedChange` from treeCheckout.ts.
 				 */
 				getChange(): JsonCompatibleReadOnly;
+				/**
+				 * Optional label provided by the user when commit was created.
+				 * This can be used by undo/redo to group or classify edits.
+				 */
+				label?: unknown;
 		  }
 		| {
 				readonly isLocal: false;
 				readonly getChange?: undefined;
+				label?: unknown;
```

Comment:
We don't want a label here, right? Since there are never labels for remote changes. Let's get some more mileage out of this discriminated union. You can either remove the property (users will have to check `isLocal` before they can check the label), or you can make it `label?: never`. The latter is probably more user friendly because of not having to check for isLocal.